### PR TITLE
1.45.1 post

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
Note that this is scheduled to go out later *today* -- it got lost in the shuffle earlier this week, so I've just written this up. Feedback greatly appreciated!

r? @rust-lang/release @rust-lang/core

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
I plan to go ahead at 1pm EDT today with this.

File: posts/2020-07-30-Rust-1.45.1.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## What's in 1.45.1 stable
+
+1.45.1 contains a collection of fixes, including one soundness fix. All patches
+in 1.45.1 affect only the 1.45.0 release; prior releases are not affected by the
+bugs this fixes.
+
+### Fix const propagation with references
+
+In Rust 1.45.0, `rustc`'s const propagation pass did not look properly handle
+encountering references, which could lead to incorrect behavior.
```
Comment (CryZe) [inline posts/2020-07-30-Rust-1.45.1.md:35]: look properly handle encountering?

Comment (Mark-Simulacrum) [reply posts/2020-07-30-Rust-1.45.1.md]: Fixed, thanks!

File: posts/2020-07-30-Rust-1.45.1.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+1.45.1][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1451-2020-07-30
+
+## What's in 1.45.1 stable
+
+1.45.1 contains a collection of fixes, including one soundness fix. All patches
+in 1.45.1 affect only the 1.45.0 release; prior releases are not affected by the
+bugs this fixes.
```
Comment (BatmanAoD) [inline posts/2020-07-30-Rust-1.45.1.md:30]:
```suggestion
1.45.1 contains a collection of fixes, including one soundness fix. All patches
in 1.45.1 address bugs that affect only the 1.45.0 release; prior releases are
not affected by the bugs fixed int his release.
```

File: posts/2020-07-30-Rust-1.45.1.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+### Fix const propagation with references
+
+In Rust 1.45.0, `rustc`'s const propagation pass did not properly handle
+encountering references when determining whether to propagate a given constant,
+which could lead to incorrect behavior.
```
[the remark this answers is not in the document]
Comment (Mark-Simulacrum) [reply posts/2020-07-30-Rust-1.45.1.md:36]: I feel somewhat like I *don't* want to call it out. It sounds horrible but really isn't all that bad.

[the remark this answers is not in the document]
Comment (BatmanAoD) [reply posts/2020-07-30-Rust-1.45.1.md:36]: I suppose that's fair, though I also think that it's good to be transparent about these things.

[the remark this answers is not in the document]
Comment (BatmanAoD) [reply posts/2020-07-30-Rust-1.45.1.md:36]: We could also mention that Cargo's failure to catch this indicates its rarity in real-world code.

[the remark this answers is not in the document]
Comment (Mark-Simulacrum) [reply posts/2020-07-30-Rust-1.45.1.md:36]: Including a snippet about crater sounds great to me.

[the remark this answers is not in the document]
Comment (Mark-Simulacrum) [reply posts/2020-07-30-Rust-1.45.1.md]: Done.

**Pull request merged**